src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Properties/InterceptedProjectProperties/ApplicationPropertyPage/ApplicationFrameworkPropertiesValueProvider.cs

```diff
@@ -0,0 +1,82 @@
+internal class ApplicationFrameworkPropertiesValueProvider : InterceptingPropertyValueProviderBase
+{
+    private const string ApplicationFrameworkProperty = "UseApplicationFramework"; //TODO: this is saved as MySubMain
+    private const string EnableVisualStylesProperty = "EnableVisualStyles";
+    private const string SingleInstanceProperty = "SingleInstance";
+    private const string SaveMySettingsOnExitProperty = "SaveMySettingsOnExit";
+    private const string HighDpiModeProperty = "HighDpiMode";
+    private const string AuthenticationModeProperty = "AuthenticationMode";
+    private const string ShutdownModeProperty = "ShutdownMode";
+    private const string SplashScreenProperty = "SplashScreen";
+    private const string MinimumSplashScreenDisplayTimeProperty = "MinimumSplashScreenDisplayTime";
+
+    private readonly IMyAppXamlFileAccessor _myAppXamlFileAccessor;
+
+    [ImportingConstructor]
+    public ApplicationFrameworkPropertiesValueProvider(IMyAppXamlFileAccessor MyAppXamlFileAccessor)
+    {
+        _myAppXamlFileAccessor = MyAppXamlFileAccessor;
+    }
+
+    public override Task<string> OnGetEvaluatedPropertyValueAsync(string propertyName, string evaluatedPropertyValue, IProjectProperties defaultProperties)
+    {
+        return GetPropertyValueAsync(propertyName);
+    }
+
+
+    public override Task<string> OnGetUnevaluatedPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties)
+    {
+        return GetPropertyValueAsync(propertyName);
+    }
+
+    public override Task<string?> OnSetPropertyValueAsync(string propertyName, string unevaluatedPropertyValue, IProjectProperties defaultProperties, IReadOnlyDictionary<string, string>? dimensionalConditions = null)
+    {
+        return propertyName switch
+        {
+            ApplicationFrameworkProperty => _myAppXamlFileAccessor.SetMySubMainAsync(unevaluatedPropertyValue),
+            SingleInstanceProperty => _myAppXamlFileAccessor.SetSingleInstanceAsync(Convert.ToBoolean(unevaluatedPropertyValue)),
+            SaveMySettingsOnExitProperty => _myAppXamlFileAccessor.SetSaveMySettingsOnExitAsync(Convert.ToBoolean(unevaluatedPropertyValue)),
+            HighDpiModeProperty => _myAppXamlFileAccessor.SetHighDpiModeAsync(Convert.ToInt16(unevaluatedPropertyValue)),
+            AuthenticationModeProperty => _myAppXamlFileAccessor.SetAuthenticationModeAsync(Convert.ToInt16(unevaluatedPropertyValue)),
+            ShutdownModeProperty => _myAppXamlFileAccessor.SetShutdownModeAsync(Convert.ToInt16(unevaluatedPropertyValue)),
+            SplashScreenProperty => _myAppXamlFileAccessor.SetSplashScreenAsync(unevaluatedPropertyValue),
+            MinimumSplashScreenDisplayTimeProperty => _myAppXamlFileAccessor.SetMinimumSplashScreenDisplayTimeAsync(Convert.ToInt16(unevaluatedPropertyValue)),
```

Comment:
These `Convert.To*` methods will throw if the input string isn't convertible. Is there any chance a non-conforming string would be received here, and if so what should we do about it?

Reply:
I see that happening on the Get method, when a property is not defined in the myapp file yet; but for the Set method, I don't see that happening (but maybe I'm wrong).